packages/ui/Marquee/Marquee.module.less

```diff
@@ -15,9 +15,50 @@
 	overflow: hidden;
 	text-align: left;
 
+	&.fade {
+		// fade left side only
+		-webkit-mask-image: -webkit-gradient(
```

Comment:
Do you mind if we simplify these?
```
	&.fade {
		&.onLeft {
			-webkit-mask-image: linear-gradient(to right, transparent, black 48px);
		}
		&.onRight {
			-webkit-mask-image: linear-gradient(to left, transparent, black 48px);
		}
		&.onLeft.onRight {
			-webkit-mask-image: linear-gradient(to right, transparent 0, black 48px, black calc(100% - 48px), transparent 100%);
		}
	}
```
Also, the current implementation uses percentages for the fade, when they likely should be fixed length.

Not tieing this to RTL as a concept is probably safer as well.

Reply:
I've updated the gradient syntax. Thanks for the suggestion.

I want to think a bit more on the RTL question. Generally, I agree with your sentiment. In this case, `MarqueeBase` accepts an `rtl` prop and tying the fade to the text directionality makes sense to me given the context.

We'd also need a way to control the size of the fade from a theme. Couple options that come to mind:

* Don't implement the fade in UI at all and leave it to the theme
* Implement the fade in UI but require that themes override it if they want a different configuration
* Use a CSS variable which can be overridden by a theme
* Use a prop on `MarqueeBase` which would likely feed into a CSS variable.

Reply:
Options 2 and 3 sound the best to me. There's also potentially a fifth option, which would be a customizable LESS variable, but with our packaging scheme, that's not settable by most users.
A sixth option could be to export a mixin from ui which generates these gradients and a theme can just use it to help options 1 and 2.